pkg/validate/validate.go

```diff
@@ -0,0 +1,215 @@
+	VSphereMachineStatus["VCenter"] = vcstatus
+
+	// If vcstatus was OK then continue & Instantiate a finder object for the client
+	if client != nil {
+
+		// Instantiate a finder object for the client
+		f := find.NewFinder(client.Client, true)
+
+		dcObject, dcvalidate, _ := DatacenterValidate(ctx, f, amachine.Datacenter)
+		VSphereMachineStatus["Datacenter"] = dcvalidate
+
+		// Now that DC is verified set the DC for finder object for all further object searches
+		f.SetDatacenter(dcObject)
+
+		netstatus, _ := NetworkValidate(ctx, f, amachine.Network.Devices[0].NetworkName)
+		VSphereMachineStatus["Network"] = netstatus
+
+		templatestatus, _ := TemplateValidate(ctx, f, amachine.Template)
+		VSphereMachineStatus["Template"] = templatestatus
+
+		return VSphereMachineStatus
+	}
+	// If you cant connect to VC then all objects should return FailureMessage
+	VSphereMachineStatus["VCenter"] = FailureMessage
+	VSphereMachineStatus["Datacenter"] = FailureMessage
+	VSphereMachineStatus["Network"] = FailureMessage
+	VSphereMachineStatus["Template"] = FailureMessage
+	return VSphereMachineStatus
+}
+
+// CheckVSphereClusterSpec Runs govmomi checks against vsphere objects that are fields of VSphereClusterSpec
+func CheckVSphereClusterSpec(acluster v1alpha2.VSphereClusterSpec) map[string]string {
+	// Creating the connection context for all API calls to VC
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Create MAP to store Connection results for all objects
+	var VSphereClusterStatus = make(map[string]string)
+
+	// Call Vcvalidate func and return client object.
+	client, vcstatus, _ := VcValidate(ctx, acluster.CloudProviderConfiguration.Global.Username, acluster.CloudProviderConfiguration.Global.Password, acluster.Server)
+	VSphereClusterStatus["VCenter"] = vcstatus
+
+	if client != nil {
+
+		// Instantiate a finder object for the client
+		f := find.NewFinder(client.Client, true)
+
+		dcObject, dcvalidate, _ := DatacenterValidate(ctx, f, acluster.CloudProviderConfiguration.Workspace.Datacenter)
+		VSphereClusterStatus["Datacenter"] = dcvalidate
+
+		// Now that DC is verified set the DC for finder object for all further object searches
+		f.SetDatacenter(dcObject)
+
+		datastorestatus, _ := DatastoreValidate(ctx, f, acluster.CloudProviderConfiguration.Workspace.Datastore)
+		VSphereClusterStatus["Datastore"] = datastorestatus
+
+		rpstatus, _ := ResourcePoolValidate(ctx, f, acluster.CloudProviderConfiguration.Workspace.ResourcePool)
+		VSphereClusterStatus["ResourcePool"] = rpstatus
+
+		folderstatus, _ := FolderValidate(ctx, f, acluster.CloudProviderConfiguration.Workspace.Folder)
+		VSphereClusterStatus["VMFolder"] = folderstatus
+
+		return VSphereClusterStatus
+	}
+	// If you cant connect to VC then all VsphereCluster objects should return FailureMessage
```

Comment:
Does this also occur when there's a transient network failure? Marking every machine in the cluster as failed as a result would be pretty destructive in that case.